### DISCUSSION TITLE
Xdebug will not update its breakpoints automatically unless you restart your application

### DISCUSSION THREAD

**Discussion by odahcam** [reactphp/reactphp#581] [Q&A]
Everytime I'm debugging and I want a new breakpoint somewhere, I have to restart the ReactPHP process so when it runs, Xdebug get to update its breakpoint references, otherwise the new breakpoint will not take effect. How do you handle that, is there a way to tell Xdebug to update the breakpoints manually?

**Comment by PaulRotmann** [comment]
It is correct that currently, one must set the breakpoints before starting the debugging process. However, this is not a limitation of ReactPHP, but rather of Xdebug itself. I am aware that this issue has been discussed in conversations with Derick Rethans. Here is also a Xdebug update where that issue is mentioned: [Xdebug Update: October 2023 — Derick Rethans](https://derickrethans.nl/xdebug-update-october-2023.html). Unfortunately, it is still not clear when this problem will be addressed.
